src/frontend/planner_test/tests/testdata/temporal_join.yaml

```diff
@@ -0,0 +1,73 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- sql: |
+    create table stream(id int, a int, b int) APPEND ONLY;
+    create table version(id int, a int, b int, primary key (id));
+    select stream.id as id1, stream.a as a1, version.id as id2, version.a as a2 from stream left join version FOR SYSTEM_TIME AS OF NOW() on stream.id = version.id
+  stream_plan: |
+    StreamMaterialize { columns: [id1, a1, id2, a2, stream._row_id(hidden)], pk_columns: [stream._row_id, id2, id1], pk_conflict: "no check" }
+    └─StreamTemporalJoin { type: LeftOuter, predicate: stream.id = version.id, output: [stream.id, stream.a, version.id, version.a, stream._row_id] }
+      ├─StreamExchange { dist: HashShard(stream.id) }
+      | └─StreamTableScan { table: stream, columns: [stream.id, stream.a, stream._row_id], pk: [stream._row_id], dist: UpstreamHashShard(stream._row_id) }
+      └─StreamNoShuffleExchange { dist: UpstreamHashShard(version.id) }
+        └─StreamTableScan { table: version, columns: [version.id, version.a], pk: [version.id], dist: UpstreamHashShard(version.id) }
+  batch_error: |-
+    Not supported: do not support temporal join for batch queries
+    HINT: please use temporal join in streaming queries
+- sql: |
+    create table stream(id int, a int, b int) APPEND ONLY;
+    create table version(id int, a int, b int, primary key (id));
+    select stream.id as id1, stream.a as a1, version.id as id2, version.a as a2 from stream join version FOR SYSTEM_TIME AS OF NOW() on stream.id = version.id where version.a < 10;
```

Comment:
It might be better to add some comments about the intention of the different test cases